dbstore_test.go

```diff
@@ -298,6 +300,91 @@ func (s *DbZuite) TestUpdateOnUpdateDoesNothing() {
 	require.Equal(s.T(), 2, ws.Version())
 }
 
+func (s *DbZuite) TestUpdateDetectsConcurrentModifications() {
```

Comment:
@alexandrinaw Realized this was not tested, and since I'm relying on this to force a failed update, chose to test it as part of this change.